tools/codegen/api/types.py

```diff
@@ -117,6 +180,9 @@ def cpp_type(self, *, strip_ref: bool = False) -> str:
         # Do not pass `strip_ref` recursively.
         return f'std::array<{self.elem.cpp_type()},{self.size}>'
 
+    def cpp_type_remove_namespaces(self) -> str:
+        return f'std::array<{self.elem.cpp_type_remove_namespaces()}, {self.size}>'
```

Comment:
The "remove" here is a bit too verb-y for me, it implies you are removing the namespaces from the CType (which you are not). Also, since this is a hack for a very specific situation (registration declarations), it is better to name the method off of how it is intended to be used, not what it does (because the functionality of a hack doesn't always make sense, but the motivation is always useful to know)

Reply:
The name isn't even accurate, you don't remove *all* namespaces, you only remove the `at::` namespace

Reply:
fair point. It's not "remove namespaces", since we don't remove `std::`. , it's "make things fit the specific format of RegistrationDeclarations.yaml. How about `def cpp_type_registration_declarations()`.